### PR TITLE
Assembler context menus for instruction

### PR DESCRIPTION
**Pull request by Nowilltolife**
## What's new

- Added context menus for instructions

## What's fixed

- Bugs 🎲 🤔 
- Annotations not showing up on field
- Signature not showing up at all
- Lables not wrapped

### CODE REVIEW

File: recaf-assembler/src/main/java/me/coley/recaf/assemble/ast/BaseElement.java [hunk 1]
```diff
@@ -149,6 +149,15 @@ public Element getChildAt(int line, int column) {
 
 	}
 
+	public Element getChildAt(int position) {
+		for (Element child : getChildren()) {
+			if (child.getStart() <= position && child.getStop() >= position) {
```
Comment (Col-E) [inline recaf-assembler/src/main/java/me/coley/recaf/assemble/ast/BaseElement.java:154]: Incredibly nit-picky, but I think `pos >= start && pos <= stop` is more immediately legible than `start <= pos && stop >= pos`.

Supporting evidence:
 - My sleep deprived brain was confused at first glance here

File: recaf-assembler/src/main/java/me/coley/recaf/assemble/ast/BaseElement.java [hunk 2]
```diff
@@ -149,6 +149,15 @@ public Element getChildAt(int line, int column) {
...
+				return child;
+			}
+		}
```
Comment (Col-E) [inline recaf-assembler/src/main/java/me/coley/recaf/assemble/ast/BaseElement.java:157]: Optimization: Can use binary search or predictive search on `Element`'s positions to skip over most elements.

File: recaf-assembler/src/main/java/me/coley/recaf/assemble/pipeline/AssemblerPipeline.java [hunk 3]
```diff
@@ -259,6 +260,10 @@ public Unit getUnit() {
 		return unit;
 	}
 
+	public List<Group> getLatestAst() {
```
Comment (Col-E) [inline recaf-assembler/src/main/java/me/coley/recaf/assemble/pipeline/AssemblerPipeline.java:263]: Will look over usage later, but having this being _"latest ast"_ may be confusing when comparing to _"unit"_ which is also occasionally referred to as the AST. Could make the naming convention of Jasm AST more specific to differentiate from Recaf AST?

File: recaf-ui/src/main/java/me/coley/recaf/ui/control/code/bytecode/AssemblerArea.java [hunk 4]
```diff
@@ -43,467 +49,637 @@
...
+import static me.coley.recaf.ui.util.Menus.action;
+import static me.darknet.assembler.parser.Group.GroupType;
 
 /**
  * Text editing portion of the assembler UI.
  *
  * @author Matt Coley
  */
 public class AssemblerArea extends SyntaxArea implements MemberEditor,
-		AstValidationListener, BytecodeValidationListener, ParserFailureListener, BytecodeFailureListener {
```
Comment (Col-E) [inline recaf-ui/src/main/java/me/coley/recaf/ui/control/code/bytecode/AssemblerArea.java:59]: Your auto-format is set to use spaces, where the rest of Recaf's source is formatted with tabs. The diff here doesn't need to be this large.

**Review by Col-E — CHANGES_REQUESTED** [review]
Some questions to address.

Minor: Some missing docs and a few formatting things. I can fix those myself though.

File: recaf-assembler/src/main/java/me/coley/recaf/assemble/pipeline/AssemblerPipeline.java (hunk 3 above)
Comment (Nowilltolife) [reply recaf-assembler/src/main/java/me/coley/recaf/assemble/pipeline/AssemblerPipeline.java:263]: i thought about Parser AST

File: recaf-ui/src/main/java/me/coley/recaf/ui/control/code/bytecode/AssemblerArea.java (hunk 4 above)
Comment (Nowilltolife) [reply recaf-ui/src/main/java/me/coley/recaf/ui/control/code/bytecode/AssemblerArea.java:59]: yes i have noticed

**Pull request merged**